### DISCUSSION TITLE
Is it possible to somehow interrupt a macro during execution?

### DISCUSSION THREAD

**Discussion by MadnessOverflow** [chhoumann/quickadd#616] [Q&A]
I am using the [Moves & Series Script](https://quickadd.obsidian.guide/docs/Examples/Macro_MovieAndSeriesScript) from the QuickAdd examples.

If a movie isn't found the script throws an error but the template after the script still gets executed, resulting in a prompt asking me for a file name (because the script didn't provide one).
Is it possible to interrupt the quickadd macro if an error occurs in the script? This would remove the unnecessary pop-up from the template.

![image](https://github.com/chhoumann/quickadd/assets/40174646/bed14580-7407-456b-80ce-c4d638ce013a)

**Comment by chhoumann** [comment]
Hey @MadnessOverflow - simply canceling should work. It doesn't cancel the entire macro, but should cancel the individual step. Macro execution is one of the things I'd like to handle better for sure.

**Comment by chhoumann** [comment]
Following up on this properly, years later. Macro interruption shipped in QuickAdd 2.10.0:

- An uncaught error in a user script now aborts the whole macro, so the template step after the movie script won't run anymore (no more file name prompt when the lookup fails).
- Scripts can also stop the macro deliberately with `params.abort("Movie not found")`, which skips all remaining steps and shows the message as a notice instead of an error dialog.

Docs: https://quickadd.obsidian.guide/docs/Choices/MacroChoice (see "Macro Execution Control").

Closing as resolved.
